UserManagementService.Application/Users/Commands/DeleteUserCommand.cs

```diff
@@ -21,8 +22,10 @@ public DeleteUserCommandHandler(IUserRepository userRepository)
         public async Task Handle(DeleteUserCommand request)
         {
             var user = await _userRepository.FindByIdAsync(request.Id);
-
-            await _userRepository.RemoveAsync(user);
+            user.Delete(
+                SystemClock.Instance.GetCurrentInstant().InUtc()

```

Comment:
We need to use https://nodatime.org/2.2.x/api/NodaTime.IClock.html for get current UTC time